lang/syntax/src/ast/ident.rs

```diff
@@ -9,3 +11,75 @@ pub type Ident = String;
 pub struct MetaVar {
     pub id: u64,
 }
+
+/// Two-dimensional De-Bruijn index
+///
+/// The first component counts the number of binder lists in scope between the variable
+/// and the binder list it originated from.
+/// The second component counts the number of binders in that binder list between the end
+/// of the binder list and the binder this variable originated from.
```

Comment:
One example for a context and the corresponding De-Bruijn index/level for an element of that context would be helpful to illustrate the concept.

Maybe we should also mention why both concepts are being used: Indices count inside-out and are stable under substituting closed terms. They are not stable under context weakening. Hence they are used for variables in the syntax tree. Levels count outside-in and are stable under context weakening. Hence they are used for addressing context elements.
Therefore, when looking up a variable from within a syntax tree in a context, a translation from indices to levels takes place.

Reply:
I have added some documentation in e3e4757 Can you check whether I am counting correctly and don't make any off-by-one mistakes?

Reply:
Looks correct to me, besides some typos.